core/src/main/java/org/mule/runtime/core/internal/routing/ScatterGatherRouter.java

```diff
@@ -47,12 +48,15 @@
 
   @Override
   protected Consumer<InternalEvent> onEvent() {
-    return event -> {
-      validateMessageIsNotConsumable(event.getMessage());
-      if (isEmpty(routes)) {
-        propagateWrappingFatal(new RoutePathNotFoundException(noEndpointsForRouter(), null));
-      }
-    };
+    return event -> validateMessageIsNotConsumable(event.getMessage());
+  }
+
+  @Override
+  public void initialise() throws InitialisationException {
+    super.initialise();
+    if (routes.size() < 2) {
```

Comment:
This was a bug in my implementation, In 3.x minimum was two, also schema enforces 2.